pkg/cmd/codespace/ssh.go

```diff
@@ -379,17 +538,25 @@ func (a *App) printOpenSSHConfig(ctx context.Context, opts sshOptions) (err erro
 	t, err := template.New("ssh_config").Parse(heredoc.Doc(`
 		Host cs.{{.Name}}.{{.EscapedRef}}
 			User {{.SSHUser}}
-			ProxyCommand {{.GHExec}} cs ssh -c {{.Name}} --stdio
+			ProxyCommand {{.GHExec}} cs ssh -c {{.Name}} --stdio -- -i {{.AutomaticIdentityFilePath}}
```

Comment:
if I understand it correctly, we don't need this line of change.
it seems `printOpenSSHConfig` won't generate or upload the key, and this `-i` arg will disable the automatic key generation as well. so if the user run `gh cs ssh --config` and `ssh` consecutively, the auth will fail, right?
also ssh args are not used when `--stdio` is present, this can be confusing.

i'd suggest we generate and upload keys when generating the config instead of when this `--stdio` command is executed. otherwise there can be a bad case where `automaticKeySSHKeysExist` being false. `hasUploadedPublicKeyForConfig` being true, and when the user is initiating frequent connections, for example when autocompleting remote file paths for `scp` command.

Reply:
This piece was added for the scenario of

```
gh cs ssh --config > ~/.ssh/config

# Change the IndentityFile to a custom key...

gh cs ssh -c :name --profile :config-profile

```

where the user has proper ssh keys already set up (i.e. doesn't need the automatic key).

In that case, the directly called `gh cs ssh` would (correctly) not generate the key, but the ProxyCommand that gets invoked *would* generate it, because it wouldn't have the same context as the original call. Adding the `-- -i` here helps ensure the proxied command also gets the correct context. 

That said, I'm not a big fan of this either and am open to suggestions. I'm not entirely even sure if the scenario above is something we need to support.

Reply:
> but the ProxyCommand that gets invoked would generate it

this doesn't seem true for me, according to line 248 to 251. did I miss anything else? :O

Reply:
The check for `-i` on 248 will skip the automatic keys, but only if `-i` is specified in the proxy command, which is why I added it here 😄 without the flag in the proxy command it would generate it

Reply:
you might misunderstand my comments. I was talking about this scenario:

```
gh cs ssh --config -c my-space > ~/.ssh/config # (a)
ssh cs.my-space.my-branch # (b) auth will fail and automatic key is not generated or uploaded.
gh cs ssh -c my-space --profile cs.my-space.my-branch # (c) this works though
```

and my proposal is that we generate and upload key (automatic key or user-specified from command line arg) in step (a) and skip the automatic key stuff altogether when `--stdio` is present.

Reply:
@Qusic thanks for the clarification, I agree that flow is currently broken if the automatic key hasn't been generated. My concern with generating the key in `--config` is that it doesn't have all the context needed to decide if the automatic key is needed or not. For example, in step `(b)` the user could specified `-- -i custom-key` at which point the automatic key wouldn't have been needed. Similarly, after step `(a)` they could edit `~/.ssh/config` and add an `IdentityFile`.

A pretty simple alternative would be to have an additional check inside the `-i` short circuit code on 248 which also checks if the supplied key is the automatic key (i.e. what `--config` would write). If so, we can go back to the normal automatic key generation route which would check if the key exists and create it if not.

Reply:
thanks for the update! looks good for me now. if automatic key exists locally, no github apis are called, so that scenarios like scp remote path completion wont be slowed down as well.

Reply:
So, um, are there any plans to unbreak the followin scenario?

> you might misunderstand my comments. I was talking about this scenario:
> 
> ```
> gh cs ssh --config -c my-space > ~/.ssh/config # (a)
> ssh cs.my-space.my-branch # (b) auth will fail and automatic key is not generated or uploaded.
> ```

Seems like doing `gh cs ssh --config` followed by `ssh cs.my-space.my-branch` is the most obvious usage of this, but it doesn't work, because `gh cs ssh --config` hardcodes usage of the `~/.ssh/codespaces.auto` which isn't being generated.